### ISSUE TITLE
Replace jQuery with Vue.js

### ISSUE THREAD

**Issue by PatrykPorabik** [grandnode/grandnode#651]
As soon as Bootstrap will drop support for jQuery, we will implement the Vue.js instead of jQuery.

**Comment by bhargava-raju** [comment]
Expressing my interest in this js migration from jQuery to Vue.js. I will work on this.

**Comment by PatrykPorabik** [comment]
Hi,
That's fantastic, but we have to discuss this changes. It would be a milestone change for the project, so we will be forced to analyze it carefully. As soon as we start to discuss it, I will ask you for feedback here.

**Comment by bhargava-raju** [comment]
looking forward to working with @PatrykPorabik and team on this.

**Comment by cedricradicia** [comment]
Just to add my point of view on this point, vue.js will not be a simple remplacement of jquery.
Vue.js is really design to develop front application, this is a really strong and performant tool to do it.
I suggest to the team to take a look on quasar-framework which is using vue.js in the good way to make desktop, mobile and web app, pwa... in fact quasar is for the front what grandnode is for the ecommerce ;)
Webpack, styling and great components library are already included (but not only)
Of course to do it, a work need to be done to separate back from front with openapi/swagger as it already done if I remember well some couple of commits.
I think a good way to integrate a new front is to use something like swaggerjs to use backend model and generate some part of the front to only have to do some UI things
By the way, the complexity could be done in backend or frontend more easily depending if each use case.

**Comment by ScratZa** [comment]
Hi Everyone - not sure if this is the correct place to post this however we are planning on integrating our front end application with GrandNode - in order to do this we would like our users to register vendors from our existing application written in angular. Am I correct in saying that we will need to extend the API to allow this functionality - only operations I can see for vendors is to get data currently. Apologies if this is a bit of a silly question

**Comment by KrzysztofPajak** [comment]
Yes, you need to extend API to create/update vendor, at this moment we only support get. But we can consider to do this for the next version.

**Comment by dogo-mehmetcantas** [comment]
Any updates for this request?

**Comment by cedricradicia** [comment]
Two new branches were created to try vuejs during last week if I'm good.
One call issue-vue-theme, the other issue-npm-vue... I can imagine this is on track but very new 😉

**Comment by dogo-mehmetcantas** [comment]
I think 5.0 or later versions is gonna be legendary 👍  Thank you

**Comment by cedricradicia** [comment]
I hope so!
This will be a big added value in comparison to SimplCommerce, VirtoCommerce and all others actors I have seen... but there are also an huge work for sure.

This is a good match with Vue3 KeyNote 😉

**Comment by dogo-mehmetcantas** [comment]
Yeah, if you need any support for backend operations please feel free to contact me. I am ready to support :)

**Comment by PatrykPorabik** [comment]
Exactly, we are working on that. You can track the issue-vue-theme branch to check how to new theme based on Vue work.
